src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,101 @@
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">6</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">
-                qui ullam ratione quibusdam voluptatem quia omnis
-              </p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">7</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">
-                illo expedita consequatur quia in
-              </p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">8</td>
+  const todos = useAppSelector(state => state.todos) || [];
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const filter = useAppSelector(state => state.filter);
+  const dispatch = useAppDispatch();
+
+  const filterByStatus = (): Todo[] => {
+    switch (filter.status) {
+      case Filters.ALL:
+        return todos;
+      case Filters.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+      case Filters.COMPLETED:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  };
+
+  const filterByQuery = (filtered: Todo[]): Todo[] => {
+    const query = filter.query.toLowerCase().trim();
+
+    return query
+      ? filtered.filter(todo => todo.title.toLowerCase().includes(query))
+      : filtered;
+  };
+
+  const filteredTodos = filterByQuery(filterByStatus());
 
+  return (
+    <table className="table is-narrow is-fullwidth">
+      <thead>
+        <tr>
+          <th>#</th>
+          <th aria-label="icon">
+            <span className="icon">
+              <i className="fas fa-check" />
+            </span>
+          </th>
+          <th>Title</th>
+          <th aria-label="empty header"> </th>
+        </tr>
+      </thead>
+
+      <tbody>
+        {filteredTodos.map(todo => (
```

Comment:
Hmm, probably I would recommend creating a separate component for `<Todo />` here, just to avoid overcomplicating this component.